MedallionShell/Shell.cs

```
@@ -71,6 +72,68 @@ public Command Run(string executable, IEnumerable<object> arguments = null, Acti
             return command;
         }
 
+        /// <summary>
+        /// Tries to attach to the already running process, given its ID.
+        /// </summary>
+        /// <param name="processId">ID of the process to attach to.</param>
+        /// <param name="attachedCommand"><see cref="Command"/> that represents the process.</param>
+        /// <returns>True if attaching was successful, false otherwise.</returns>
+        public bool TryAttachToProcess(int processId, out Command attachedCommand)
+        {
+            return this.TryAttachToProcess(processId, options => { }, out attachedCommand);
+        }
+
+        // Parameter has no matching param tag in the XML comment (but other parameters do)
+        /// <summary>
+        /// Tries to attach to the already running process, given its ID and additional options.
+        /// </summary>
+        /// <inheritdoc cref="TryAttachToProcess(int,out Medallion.Shell.Command)"/>
+        /// <param name="options">Additional options when attaching to the process.</param>
+#pragma warning disable CS1573 // Reason: the parameters issued in the warning are actually inherited.
+        public bool TryAttachToProcess(int processId, Action<Shell.Options> options, out Command attachedCommand)
+#pragma warning restore CS1573
+        {
+            var finalOptions = this.GetOptions(options);
```

We should throw `InvalidOperationException` here if `StartInfo` or `Encoding` has been specified on the options since we can't respect those as it is too late to change them.

---

Done.